spec/requests/api/v0/highlights_controller_spec.rb

```diff
@@ -439,10 +439,53 @@
     end
   end
 
+  context 'GET /highlights/summary' do
+    let!(:highlight1) { create(:highlight, id: fake_uuid(1), user_uuid: user_uuid, source_id: source_id, scope_id: scope_1_id) }
+    let!(:highlight2) { create(:highlight, id: fake_uuid(2), user_uuid: user_uuid,                       scope_id: scope_1_id) }
+    let!(:highlight3) { create(:highlight, id: fake_uuid(3), user_uuid: user_uuid,                       scope_id: scope_1_id, color: "red") }
+    let!(:highlight4) { create(:highlight, id: fake_uuid(4), user_uuid: user_uuid, source_id: source_id, scope_id: scope_1_id, prev_highlight: highlight1) }
+    let!(:highlight5) { create(:highlight, id: fake_uuid(5), user_uuid: user_uuid, source_id: source_id, scope_id: scope_1_id, prev_highlight: highlight1, next_highlight: highlight4) }
+    let!(:highlight6) { create(:highlight, id: fake_uuid(6), user_uuid: user_uuid, source_id: source_id, scope_id: SecureRandom.uuid) }
+    let!(:highlight7) { create(:highlight, id: fake_uuid(7)) }
+
+    let(:scope_id) { highlight1.scope_id }
+    let(:query_params) do
+      {
+        source_type: 'openstax_page',
+        scope_id: scope_id,
+        color: 'yellow',
+      }
+    end
+
+    context 'when a user is logged in' do
+      before { stub_current_user_uuid(user_uuid) }
+
+      it 'returns good counts' do
+        get summary_path, params: query_params
+        expect(response).to have_http_status(:ok)
+        expect(json_response[:counts_per_source]).to eq({
+          "#{source_id}".to_sym => 3,
```

Comment:
am i seeing this wrong?  looks like there should be 4 HLs for source_id and user_uuid, one for each of 2 & 3.

Reply:
should this show some examples of filtering and HL counts, across a few users, source_id, source_type, and color variations?

Reply:
If we change highlight 5 to have a different user then we’d be testing filtering by user better.  Already tests color variation (there’s the red one in there not counted).  We could add a HL with a different source_type.  There’s no filtering by source_id on the summary endpoint

Reply:
added testing of filtering by user, can't test filtering by source_type b/c there is only one allowed source_type